### ISSUE TITLE
Video playback occasionally get stuck in this reproducer

### ISSUE THREAD

**Issue by Vizalkar** [wang-bin/mdk-sdk#179]
**Bug description**
Some videos get stuck after going from paused state to playing state
This happens in our commercial software and I reproduced it in this [example project](https://drive.google.com/file/d/1KPQhwsCC8wFJeQ2hzRQmEFakTOC1M5CU/view?usp=sharing)
[Here](https://drive.google.com/file/d/1zhAkcksh6fV7YgRrF7IZaofX6DmHv3EC/view?usp=sharing) is a video of the reproduced bug
As you can see, it results in one of the video (the one in the selected box) being playing at a very slow framerate
NB: the OBS capture itself is lagged but it still shows the bug happening, the program was running smoothly at the time of capture. In fact OBS was started once the bug had been reproduced

**To Reproduce**
Build the Qt project attached to this issue and run the program
Proceed to the actions as shown in the video's action logs : 
- press up arrow key (adds a video) x 4
- press space bar key (switch between play and pause on all boxes) x 4

**Expected behavior**
Given the fact that the computer has enough power to play all videos simultaneously without any lag at the beginning of the example, I would expect the video to play normally when i switch from paused to playing state.

**Environment:**
 - OS: MacOS Sonoma 14.1.2 (23B92)
 - GPU: M1

**Information so that you can look into it**
The example project is a modified version of "boxes" from Qt examples.
The `mdk::Player` instances are inside the `GLTexture2DVideo` class, in `glbuffers.h`
The `QtBox` class has a `GLTexture2DVideo` that it uses for its rendering
`Scene::keyPressEvent(QKeyEvent *event)` is the function for which actions are logged
The shiny rounded boxes are not instances of the `QtBox` class, they're other objects that share the same `GLTexture2DVideo`, they are not controlled by the actions in `keyPressEvent` : At the end of the reproducer, the number of `mdk::Player` instanciated is 5

Regards

**Comment by Vizalkar** [comment]
Hi,
Do you have any updates concerning this issue ? Were you able to reproduce ? 
When can we expect it to be fixed ? 
Regards

**Comment by wang-bin** [comment]
I can reproduce on mac. I modify multiplayers example to pause resume periodicly, the bug can be reproduced in that example if I run native macos executable, no such bug if I run windows exectuable via wine. I guess the bug comes from audio renderer, if audio tracks are disabled, playback is good.

**Comment by Vizalkar** [comment]
Thank you very much for the info

**Comment by Vizalkar** [comment]
Do you have any recent updates on the subject ?
This is quite an important bug for us, we're waiting for the fix to release a patch

Respectul regards

**Comment by wang-bin** [comment]
I can't reproduce the issue in my example, but your example is still broken

**Comment by Vizalkar** [comment]
So you mean you fixed half of the bug but some cases remain ?
Will you be able to fix it entirely ?
Regards

**Comment by wang-bin** [comment]
no, i still don't know the reason, i can try another audio renderer this weekend

**Comment by wang-bin** [comment]
fixed in the latest build
